### ISSUE TITLE
S3 empty file upload causes 8MB file with TransferUtility

### ISSUE THREAD

**Issue by consulting-dev** [aws/aws-sdk-net#3136]
### Describe the bug

We have ASP.NET 6.0 application which works as a proxy between client and S3.
Recently we detected issue that if client uploads empty file (0 bytes) - in S3 it lands as 8MB file filled with <null> bytes.
![image](https://github.com/aws/aws-sdk-net/assets/155075432/eedaf685-86ec-4c5d-82ec-c2476d7e6f04)
Investigation showed that issue is caused by a stream which has CanSeek equal to false.

### Expected Behavior

Empty file should be uploaded as is.

### Current Behavior

Empty file is uploaded as 8Mb file filled with null bytes.

### Reproduction Steps

```
using Amazon;
using Amazon.S3;
using Amazon.S3.Transfer;

using Xunit;

namespace Unit.Tests;

public class SimpleStorageServiceShould
{
    [Fact]
    public async Task UploadFile()
    {
        var accessKey = "<access key value>";
        var secretKey = "<secret key value>";
        var client = new AmazonS3Client(accessKey, secretKey, new AmazonS3Config { RegionEndpoint = RegionEndpoint.EUCentral1, });

        // here is our non seakable empty stream!
        var stream = new CustomStream();

        using var fileTransferUtility = new TransferUtility(client);
        var uploadRequest = new TransferUtilityUploadRequest
        {
            Key = $"{Guid.NewGuid():N}.txt",
            BucketName = "unit-tests",
            DisableDefaultChecksumValidation = false,
            InputStream = stream
        };

        await fileTransferUtility.UploadAsync(uploadRequest, CancellationToken.None);
    }

    public class CustomStream : MemoryStream
    {
        public override bool CanSeek => false;
    }
}
```

### AWS .NET SDK and/or Package version used

AWSSDK.S3 3.7.305.3
Older versions have the same issue

### Targeted .NET Platform

.NET 6

### Operating System and version

Ubuntu, Docker

**Comment by dscpinheiro** [comment]
We're investigating this issue, we've recently modified the Transfer Utility to handle unseekable or unknown length streams (https://github.com/aws/aws-sdk-net/commit/305e5c9b70600d89cd54e4876fcef5fa277ea9e2). This was released on version `3.7.202.0` of the `AWSSDK.S3` package and we believe it may be related.

I have tested your code with an older version (`3.7.201.17`) and the file created on S3 has `0 B` as its size.
